vendor/kcfinder/core/bootstrap.php

```diff
@@ -108,6 +108,42 @@ function checkLSSession()
     {
         define("BASEPATH", realpath($sLimesurveyFolder . "/../vendor/yiisoft/yii/framework") . "/");
     }
+
+    if (!defined('YII_DEBUG')) {
+        if (file_exists(APPPATH.'config'.DIRECTORY_SEPARATOR.'config.php')) {
+            $aSettings= include(APPPATH.'config'.DIRECTORY_SEPARATOR.'config.php');
+        } else {
+            $aSettings=array();
+        }
+        
+        // Set debug : if not set : set to default from PHP 5.3
+        if (isset($aSettings['config']['debug'])) {
```

Comment:
It's a copy/paste from index ? Unsure if we need to move to a function ?

Reply:
Or maybe include a file to set debug (same for index and kcfinder (and maybe command)) `include(application/config/setdebug.php)`

Reply:
If doing this, we would need to load the config file independently for that function. Hence on the index.php the config file would be loaded twice.

Again, I believe we need some code reorg, but I believe it should be as a separate task.

Reply:
> If doing this, we would need to load the config file independently for that function. Hence on the index.php the config file would be loaded twice.

We just include the system to set debug, not load config a second time …

Just include …

> Again, I believe we need some code reorg, but I believe it should be as a separate task.

Currently : debug system is only in index, you copy paste on this commit … then it be a reason of reorganization.

Reply:
> We just include the system to set debug, not load config a second time …

But the config.php would be included twice, right? once on the general bootstrapping and then again for the setdebug.php new file, right?

But maybe already it is happening now... I will give it a thought.

Reply:
> then again for the setdebug.php new file, right?

Maye just a finction in include, using $settings array ?

Reply:
Check now please